examples/02_decoding/plot_haxby_grid_search.py

```diff
@@ -37,98 +37,99 @@
 # Load the Haxby dataset
 # -----------------------
 from nilearn import datasets
-import numpy as np
-import pandas as pd
 # by default 2nd subject data will be fetched on which we run our analysis
 haxby_dataset = datasets.fetch_haxby()
+fmri_filename = haxby_dataset.func[0]
+mask_filename = haxby_dataset.mask
 
 # print basic information on the dataset
 print('Mask nifti image (3D) is located at: %s' % haxby_dataset.mask)
 print('Functional nifti image (4D) are located at: %s' % haxby_dataset.func[0])
 
 # Load the behavioral data
+import pandas as pd
 labels = pd.read_csv(haxby_dataset.session_target[0], sep=" ")
 y = labels['labels']
-session = labels['chunks']
+
 
 # Keep only data corresponding to shoes or bottles
+from nilearn.image import index_img
 condition_mask = y.isin(['shoe', 'bottle'])
-y = y[condition_mask]
 
-###########################################################################
-# Prepare the data with the NiftiMasker
-from nilearn.input_data import NiftiMasker
-
-mask_filename = haxby_dataset.mask
-# For decoding, standardizing is often very important
-nifti_masker = NiftiMasker(mask_img=mask_filename, sessions=session,
-                           smoothing_fwhm=4, standardize=True,
-                           memory="nilearn_cache", memory_level=1)
-func_filename = haxby_dataset.func[0]
-X = nifti_masker.fit_transform(func_filename)
-# Restrict to non rest data
-X = X[condition_mask]
-session = session[condition_mask]
+fmri_niimgs = index_img(fmri_filename, condition_mask)
+y = y[condition_mask]
+session = labels['chunks'][condition_mask]
 
```

Comment:
Maybe you could also demonstrate very quickly how to grid-search the SVC parameter (or for another model)  just to show the API of grid_param, and return the best param kept ?